sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sfas-integration.scheduler.e2e-spec.ts

```diff
@@ -607,6 +615,220 @@ describe(describeProcessorRootTest(QueueNames.SFASIntegration), () => {
+            dependantBirthDate: true,
+          },
+          relations: { application: true },
+          where: { id: expectedSFASApplicationDependant.id },
+        });
+      expect(sfasApplicationDependant).toEqual(
+        expectedSFASApplicationDependant,
+      );
+
+      // Verify SFAS Application Disbursement created.
+      const expectedSFASApplicationDisbursement: Partial<SFASApplicationDisbursement> =
+        {
+          id: 1148372345,
+          application: { id: 14541 } as SFASApplication,
+          fundingType: "CSL",
+          fundingAmount: 300,
+          fundingDate: "2025-01-20",
+          dateIssued: "2025-01-30",
+        };
+      const sfasApplicationDisbursement =
+        await db.sfasApplicationDisbursement.findOne({
+          select: {
+            id: true,
+            application: { id: true },
+            fundingType: true,
+            fundingAmount: true,
+            fundingDate: true,
+            dateIssued: true,
+          },
+          relations: { application: true },
+          where: { id: expectedSFASApplicationDisbursement.id },
+        });
+      expect(sfasApplicationDisbursement).toEqual(
+        expectedSFASApplicationDisbursement,
+      );
+    },
+  );
+
+  it(
+    "Should log process summary warning and proceed to process the next record " +
+      "when the SFAS file has one or more records with an invalid record type.",
+    async () => {
+      // Arrange
+      // Queued job.
+      const mockedJob = mockBullJob<void>();
+      mockDownloadFiles(sftpClientMock, [INVALID_RECORD_TYPE_FILENAME]);
+
+      // Act
+      await processor.processQueue(mockedJob.job);
+
+      // Assert
+      // Expect the file was archived on SFTP.
+      expect(sftpClientMock.rename).toHaveBeenCalled();
+      // Expect the warning message logged for invalid record type.
+      expect(
+        mockedJob.containLogMessages([
+          "WARN: No data importer to process the record type 700 at line number 2.",
+          "File contains 2 records.",
+          "Updating student ids for SFAS individuals.",
+          "Student ids updated.",
+          "Updating and inserting new disbursement overaward balances from sfas to disbursement overawards table.",
+          "New disbursement overaward balances inserted to disbursement overawards table.",
+          "Inserting student restrictions from SFAS restrictions data.",
+          "Inserted student restrictions from SFAS restrictions data.",
+        ]),
+      ).toBe(true);
+    },
+  );
+
+  /**
+   * Delete all the legacy data for clean data execution.
+   * @param db data source.
+   */
+  async function deleteSFASData(db: E2EDataSources) {
+    await db.sfasIndividual.delete({});
+    await db.sfasApplication.delete({});
+    await db.sfasPartTimeApplications.delete({});
+    await db.sfasRestriction.delete({});
+    await db.sfasApplicationDependant.delete({});
+    await db.sfasApplicationDisbursement.delete({});
+  }
```

Comment:
Since these are tables with no FK, truncating them would be faster than deleting the records, was there a benefit to using the delete for these cases?

Reply:
No particular reason for choosing delete other than it has been more commonly used and didn't consider the truncate for the operation.

I can change to truncate if that will make considerable difference.

Reply:
As we talked not a blocker. We use the `TRUNCATE` when possible and the tables do not have FKs, not not mandatory for E2Es.